src/components/Touchable.js

```diff
@@ -61,6 +62,26 @@ export default class Touchable extends Component<
     }
   };
 
+  onLongPress = async () => {
+    const { onLongPress, event, eventProperties } = this.props;
+    if (!onLongPress) return;
+    try {
+      if (event) {
+        track(event, eventProperties);
+      }
+      const res = onLongPress();
+      if (res && res.then) {
+        // it's a promise, we will use pending/spinnerOn state
+        this.setState({ pending: true });
+        await res;
+      }
+    } finally {
+      if (!this.unmounted) {
+        this.setState(({ pending }) => (pending ? { pending: false } : null));
+      }
+    }
+  };
```

Comment:
@grsoares21 seems like this block is a bit overkill, can't we just do this and if it's not a promise then it should be fast enough to not be noticeable?? (Haven't tested it, so not my fault if it doesn't work)

```suggestion
  onLongPress = async () => {
      const { onLongPress, event, eventProperties } = this.props;
      if (!onLongPress) return;
      if (event) {
        track(event, eventProperties);
      }
      
      const res = onLongPress();
      this.setState({ pending: true });
      await res;
      this.setState({ pending: false });
  };
```